files.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const debug = require('diagnostics')('warehouse:files');
+const mime = require('mime-types');
+const fs = require('fs').promises;
+const crypto = require('crypto');
+const path = require('path');
+
+/**
+ * @const {string} HASH_ALGORITHM Hashing algorithm used for digest.
+ */
+const HASH_ALGORITHM = 'sha256';
+
+/**
+ * Generate digest for file.
+ *
+ * @param {Buffer} content File content to hash
+ * @returns {string} digest
+ * @private
+ */
+function digest(content) {
+  const hash = crypto.createHash(HASH_ALGORITHM);
+  hash.update(content);
+
+  return `${ HASH_ALGORITHM }-${ hash.digest('base64') }`;
+}
+
+/**
+ * @typedef {import('./index.js')} Warehouse
+ */
+/**
+ * @constructor
+ * @param {Warehouse} warehouse Reference to the Warehouse instance.
+ * @param {Object} config configuration
+ * @param {string} [config.root=process.cwd()] Root directory to resolve files from.
+ * @public
+ */
+class Files {
+  constructor(warehouse, { root = process.cwd() } = {}) {
+    this.warehouse = warehouse;
+    this.root = root;
+  }
+
+  /**
+   * Read the content of the files.
+   *
+   * @param {string[]} [files=[]] Files to read.
+   * @returns {Promise<Buffer[]>} File content as Buffer.
+   * @public
+   */
+  async read(files = []) {
+    const root = this.root;
+    const content = await Promise.all(files.map(file => {
+      try {
+        return fs.readFile(file);
+      } catch (error) {
+        debug(`Unable to read ${file}: ${ error.message }, resolving from ${ root }`);
+      }
+
+      try {
+        file = path.resolve(this.root, file);
+        return fs.readFile(file);
+      } catch (err) {
+        return err;
```

Comment:
Is this right? if you can't read the file, rather than having the file buffer it should have the error in the array?

Reply:
What does it mean for this API to not be able to read a file that the caller told it about? Should it not just throw/return a rejected promise? Presumably this API is for creating the attachments for uploading file contents to warehouse by manually and bypassing the build.  Should that not fail if a file isn't available rather than just silently continuing along?

Reply:
Update to just throw, this was indeed not correct.